tests/test-multi-pow.c

```diff
@@ -0,0 +1,107 @@
+/* Test program for thread-safe dcurl */
+#include <pthread.h>
+#include "common.h"
+#include "dcurl.h"
+
+#define THREAD_MAX 10
+
+typedef struct _ducrl_item DCURL_ITEM;
+struct _ducrl_item {
+    int mwm;
+    int8_t input_trytes[TRANSACTION_TRYTES_LENGTH];  /* 2673 */
+    int8_t output_trytes[TRANSACTION_TRYTES_LENGTH]; /* 2673 */
+};
+
+void *dcurl_entry_cb(void *arg)
+{
+    DCURL_ITEM *dcurl_item = (DCURL_ITEM *) arg;
+    /* test dcurl Implementation with mwm = 9 */
+    int8_t *ret_trytes =
+        dcurl_entry(dcurl_item->input_trytes, dcurl_item->mwm, 0);
+    assert(ret_trytes);
+    memcpy(dcurl_item->output_trytes, ret_trytes, TRANSACTION_TRYTES_LENGTH);
+    free(ret_trytes);
+
+    pthread_exit(NULL);
```

Comment:
I am not sure if it is safe to invoke `pthread_exit` here.

Reply:
Hmm...... isn't it the right way of terminating a thread?
Is there any potential risk?

Reply:
We might have no information how the program is terminated, that is a matter of test plan flow control.

Reply:
Eventually, we would not depend on the messages. Instead, we shall be able to check all return values to be acquainted with the state of given test items.